anta/result_manager/__init__.py

```diff
@@ -195,12 +264,15 @@ def add(self, result: TestResult) -> None:
         result
             TestResult to add to the ResultManager instance.
         """
+        # Register the manager to the result to allow status updates
+        result.register_manager(self)
+
         self._result_entries.append(result)
         self._update_status(result.result)
         self._update_stats(result)
 
         # Every time a new result is added, we need to clear the cached property
-        self.__dict__.pop("results_by_status", None)
+        self.clear_cache()
```

Comment:
the way we use it it makes it a bit many calls

Reply:
lets use a flag

Reply:
add a decorator on `result_by_status` -> if flag `stats_in_sync` is False then pop the porperty

Reply:
Fixed in 3c3e8ee3af4ff3ea57210e75835ee19d27f7105b